### PR TITLE
feat(portex): add schema update in "DataFrame.__setitem__"

### PR DESCRIPTION
**Pull request by Lee-000**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-28v2geg 支持追加列行为在 DataFrame 和 schema 之间的联动](https://app.clickup.com/t/28v2geg)

**Review by AChenQ — APPROVED** [review]

File: graviti/dataframe/frame.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def __setitem__(self, key: str, value: Union[Iterable[Any], _C]) -> None:
                 f"self DataFrame ({len(self)})"
             )
 
-        # TODO: Need edit the schema.
+        self.schema[key] = schema
```
Comment (linjiX) [inline graviti/dataframe/frame.py:155]: we need to add check for replace column

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/frame.py [hunk 2]
```diff
@@ -137,6 +137,9 @@ def __getitem__(self, key: Union[str, Iterable[str]]) -> Container:
         raise KeyError(key)
 
     def __setitem__(self, key: str, value: Union[Iterable[Any], _C]) -> None:
+        if self.operations is not None and key in self._column_names:
+            raise NotImplementedError("Replacing a column is not supported yet")
```
Comment (linjiX) [inline graviti/dataframe/frame.py:141]: do we need to mention the dataframe is a sheet in commit, not a local dataframe?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/frame.py [hunk 3]
```diff
@@ -137,6 +137,9 @@ def __getitem__(self, key: Union[str, Iterable[str]]) -> Container:
         raise KeyError(key)
 
     def __setitem__(self, key: str, value: Union[Iterable[Any], _C]) -> None:
+        if self.operations is not None and key in self._column_names:
```
Comment (linjiX) [inline graviti/dataframe/frame.py:140]: we need the check operatinos and _colmun_names again, can we refiine it?

**Review by linjiX — CHANGES_REQUESTED** [review]

**Review by linjiX — CHANGES_REQUESTED** [review]

**Pull request merged**